flax/linen/fp8_ops.py

```diff
@@ -664,3 +623,24 @@ def __call__(self, *args, **kwargs):
 class NANOOFp8DotGeneralOp(Fp8DotGeneralOp):
   e4m3_dtype: DType = jnp.float8_e4m3fnuz
   e5m2_dtype: DType = jnp.float8_e5m2fnuz
+
+class Fp8Einsum(Fp8DotGeneralBase):
+
+  def __call__(self, eqn, lhs: jnp.ndarray, rhs: jnp.ndarray) -> jnp.ndarray:
+    # Here we assume that the rhs is the weight and its dtype is the actual compute dtype (not storage dtype).
+    # TODO(kaixih@nvidia): Better way to handle this?
+    actual_compute_dtype = rhs.dtype
+    lhs = lhs.astype(actual_compute_dtype)
+
+    dot_general_fn = partial(
+        fp8_scaled_dot_general,
+        lhs_scale=self.input_scale.value,
+        rhs_scale=self.kernel_scale.value,
+        grad_scale=self.output_grad_scale.value,
+        lhs_amax_history=self.input_amax_history.value,
+        rhs_amax_history=self.kernel_amax_history.value,
+        grad_amax_history=self.output_grad_amax_history.value,
+        quantize_compute_type=actual_compute_dtype
```

Comment:
Should `Fp8Einsum` related to `Fp8DirectDotGeneralOp`? In `Fp8DirectDotGeneralOp`,  `preferred_element_type=x.dtype`, shouldn't `Fp8Einsum` follow the same setting?